src/github3/checks.py

```diff
@@ -0,0 +1,541 @@
+"""This module contains all the classes relating to Checks."""
+from __future__ import unicode_literals
+
+from json import dumps
+
+from . import decorators
+from . import models
+
+
+class CheckPullRequest(models.GitHubCore):
+    """Representation of a Pull Request returned in Checks APIs.
+
+    .. versionadded:: 1.3.0
+
+    .. note::
+
+        Refreshing this object returns a :class:`~github3.pulls.PullRequest`.
+
+    This object has the following attributes:
+
+    .. attribute:: id
+
+        The unique id of this pull request across all of GitHub.
+
+    .. attribute:: number
+
+        The number of this pull request on its repository.
+
+    .. attribute:: head
+
+        A dict of minimal head information retrieved from the Check data
+        representing the source of the pull request
+
+    .. attribute:: base
+
+        A dict of minimal base information retrieved from the Check data
+        representing the pull request destination.
+    """
+
+    def _update_attributes(self, pull):
+        self.id = pull["id"]
+        self.number = pull["number"]
+        self.base = pull["base"]
+        self.head = pull["head"]
+        self._api = self.url = pull["url"]
+
+    def _repr(self):
+        return "<CheckPullRequest [#{0}]>".format(self.number)
+
+    def to_pull(self):
+        """Retrieve a full PullRequest object for this CheckPullRequest.
+
+        :returns:
+            The full information about this pull request.
+        :rtype:
+            :class:`~github3.pulls.PullRequest`
+        """
+        from . import pulls
+
+        json = self._json(self._get(self.url), 200)
+        return self._instance_or_null(pulls.PullRequest, json)
+
+    refresh = to_pull
+
+
+class CheckApp(models.GitHubCore):
+    """Representation of an App returned in Checks APIs.
+
+    .. versionadded:: 1.3.0
+
+    .. note::
+
+        Refreshing this object returns a :class:`~github3.apps.App`.
+
+    This object has the following attributes:
+
+    .. attribute:: description
+
+        The description of the App provided by the owner.
+
+    .. attribute:: external_url
+
+        The URL provided for the App by the owner.
+
+    .. attribute:: html_url
+
+        The HTML URL provided for the App by the owner.
+
+    .. attribute:: id
+
+        The unique identifier for the App. This is useful in cases where you
+        may want to authenticate either as an App or as a specific
+        installation of an App.
+
+    .. attribute:: name
+
+        The display name of the App that the user sees.
+
+    .. attribute:: owner
+
+        A dict of minimal user information retrieved from the Check data
+        representing the app owner
+    """
+
+    def _update_attributes(self, app):
+        self.description = app["description"]
+        self.external_url = app["external_url"]
+        self.html_url = app["html_url"]
+        self.id = app["id"]
+        self.name = app["name"]
+        self.owner = app["owner"]
```

Comment:
I think we're missing `_api` here. `to_app` and `refresh` won't work unless we construct the URL from scratch from the id

Reply:
This would look like, I think

```suggestion
self._api = self.url = app["url"]
# or
self._api = self.url = self._build_url("app", str(self.id))
```

I'm not sure which is easier

Reply:
Looks like you can't get a GitHub App by id. You need the [`app_slug`](https://developer.github.com/v3/apps/#get-a-single-github-app) which is not returned from the API. GitHub doesn't seem to return a API URL for the App in the [CheckSuite API](https://developer.github.com/v3/checks/suites/#response).

Reply:
I think we can generate it from the `html_url` actually:

```
  "html_url": "https://github.com/apps/super-ci",
```

Something like:

```py
_, slug = html_url.rsplit('/', 1)
```

Reply:
Done.